fmriprep/workflows/util.py

```diff
@@ -17,12 +17,33 @@
 from niworkflows.nipype.pipeline import engine as pe
 from niworkflows.nipype.interfaces import utility as niu
 from niworkflows.nipype.interfaces import fsl, afni, ants, freesurfer as fs
-from niworkflows.interfaces.registration import FLIRTRPT, BBRegisterRPT
+from niworkflows.interfaces.registration import FLIRTRPT, BBRegisterRPT, MRICoregRPT
 from niworkflows.interfaces.masks import SimpleShowMaskRPT
 
 from ..interfaces.images import extract_wm
 
 
+def compare_xforms(fallback_mat, test_mat):
+    import numpy as np
+    from transforms3d.affines import decompose44
+    from transforms3d.axangles import mat2axangle
+    # [[R], [A], [S], [1]] = mat1.dot([[x], [y], [z], [1]])
+    mat1 = np.loadtxt(fallback_mat)
+    # [[R'], [A'], [S'], [1]] = mat2.dot([[x], [y], [z], [1]])
+    mat2 = np.loadtxt(test_mat)
+    # [[R'], [A'], [S'], [1]] = mat2.dot(mat1_inv.dot([[R], [A], [S], [1]]))
+    comp = mat2.dot(np.linalg.pinv(mat1))
+    trans, rotation_matrix, scales, shears = decompose44(comp)
+
+    max_trans = np.max(np.abs(trans))
+    rot = mat2axangle(rotation_matrix)[1]
+    max_scale = np.max(np.abs(scales))
+
+    fallback = any((max_trans > 2, rot > np.pi / 36, max_scale > 1.1))
```

Comment:
2mm is not too small?

Reply:
That might be. What seems like a reasonable translation threshold? It seems like it shouldn't be more than a couple voxels, right? (I think I picked 2 as a reasonable "1 voxel" threshold, but it was admittedly a quick and dirty guess.)